frontend/public/components/app.jsx

```diff
@@ -214,11 +214,17 @@ setInterval(() => store.dispatch(UIActions.updateTimestamps(Date.now())), 10000)
 fetchSwagger();
 
 // Used by GUI tests to check for unhandled exceptions
-window.windowError = false;
-window.onerror = window.onunhandledrejection = (e) => {
+window.windowError = null;
+window.onerror = (message, source, lineno, colno, error) => {
   // eslint-disable-next-line no-console
-  console.error('Uncaught error', e);
-  window.windowError = e || true;
+  console.error('Uncaught error', error);
+  window.windowError = error;
+};
+window.onunhandledrejection = (promiseRejectionEvent) => {
+  const error = new Error(`unhandled promise rejection: ${promiseRejectionEvent.reason}`);
```

Comment:
If this isn't already an error, maybe we just leave `window.windowError` as `Error | Event` and just check that it's not defined in Cypress rather than trying to rethrow it.

Reply:
I don't think we'd see any output for `promiseRejectionEvent` in Cypress if we did that.

Reply:
Note: previously the `window.windowError` was either an `ErrorEvent` or `promiseRejectionEvent`, it never was an `Error`.  I had to change the method signiture params in order to get the `Error`: `window.onerror = (message, source, lineno, colno, error)`.

Reply:
I do know that when `window.windowError` was a `promiseRejectionEvent`, throwing it didn't produce any output in headless.

Reply:
Right, you wouldn't throw the event. You would check that it's not null and log the content (if not already logged by Cypress).

Reply:
idk, I feel casting all into an `Error` and just handling/throwing the `Error` (directly or recreating w/ formatting), to me seems more straight forward.   Unless you are saying we should log the `promiseRejectionEvent`, not throw an Error, so that the cypress tests won't fail on a `promiseRejectionEvent`?

Reply:
I'm saying we have an assertion that causes the test to fail if `windowError` is never null. Something like

```ts
expect(win.windowError).to.not.exist;
```

Reply:
Hi, tried
```
assert.isNull(
      win.windowError,
      `${(win.windowError as PromiseRejectionEvent).reason || (win.windowError as Error).message} ${
        (win.windowError as Error).stack
      }`
```
Which, in addition to the message (2nd arg in assert(..)), also output the tested object.  Unfortunately, since `window` is part of the PromiseRejectionEvent, it gets spit out along with a lot of other attributes:
```
2) Monitoring: Alerts
       "after each" hook for "creates and expires a Silence":
     path /silence_WRONG_URL/63646b85-b4a6-4ead-8a35-609a77317edf was not found undefined: expected { isTrusted: [Getter],
  promise: {},
  reason: 'path /silence_WRONG_URL/63646b85-b4a6-4ead-8a35-609a77317edf was not found',
  NONE: 0,
  CAPTURING_PHASE: 1,
  AT_TARGET: 2,
  BUBBLING_PHASE: 3,
  type: 'unhandledrejection',
  target: 
   { parent: 
      { parent: [Circular],
        opener: null,
        top: [Circular],
        length: 2,
        frames: [Circular],
        closed: false,
        location: [Object],
        self: [Circular],
        window: [Circular],
        document: [Getter],
        name: [Getter/Setter],
        customElements: [Getter],
        history: [Getter],
        locationbar: [Getter/Setter],
        menubar: [Getter/Setter],
        personalbar: [Getter/Setter],
        scrollbars: [Getter/Setter],
        statusbar: [Getter/Setter],
        toolbar: [Getter/Setter],
        status: [Getter/Setter],
        frameElement: [Getter],
        navigator: [Getter],
        origin: [Getter/Setter],
        external: [Getter/Setter],
        screen: [Getter/Setter],
        innerWidth: [Getter/Setter],
        innerHeight: [Getter/Setter],
        scrollX: [Getter/Setter],
        pageXOffset: [Getter/Setter],
        scrollY: [Getter/Setter],
        pageYOffset: [Getter/Setter],
        visualViewport: [Getter/Setter],
        screenX: [Getter/Setter],
        screenY: [Getter/Setter],
        outerWidth: [Getter/Setter],
        outerHeight: [Getter/Setter],
        devicePixelRatio: [Getter/Setter],
        clientInformation: [Getter/Setter],
...
...
  -- additional two screens worth of output ---
```

Reply:
Latest code I think moves closer to what your after. Changed the assertion so it doesn't ouput entire contents of `win.windowError`.  It does output `...: expected false to be true` which doesn't add any value:

```
assert.isTrue(
      win.windowError === null,
      `window error detected: ${(win.windowError as PromiseRejectionEvent).reason ||
        (win.windowError as Error).message} ${
        (win.windowError as Error).stack
          ? (win.windowError as Error).stack.replace(/\\n/g, '\n')
          : ''
      }`,
    );
```
**Produces:**
```
...when windowError is a PromiseRejectionEvent:

 2) Monitoring: Alerts
       "after each" hook for "creates and expires a Silence":
     AssertionError: window error detected: path /silence_WRONG_URL/b0ef08c1-e09c-41ab-a046-761d47a71bee was not found : expected false to be true

...when windowError is an Error:

 2) Monitoring: Alerts
       "after each" hook for "creates and expires a Silence":
     window error detected: i is not defined ReferenceError: i is not defined
    at http://localhost:9000/static/main-66d801b01502734ad2c0.js:96412:5
    at renderWithHooks (http://localhost:9000/static/vendors~main-9d6cea73c651a7e7c901.js:377121:18)
    at mountIndeterminateComponent (http://localhost:9000/static/vendors~main-9d6cea73c651a7e7c901.js:379355:13)
    at beginWork$1 (http://localhost:9000/static/vendors~main-9d6cea73c651a7e7c901.js:380499:16)
    at HTMLUnknownElement.callCallback (http://localhost:9000/static/vendors~main-9d6cea73c651a7e7c901.js:362360:14)
    at Object.invokeGuardedCallbackDev (http://localhost:9000/static/vendors~main-9d6cea73c651a7e7c901.js:362410:16)
    at invokeGuardedCallback (http://localhost:9000/static/vendors~main-9d6cea73c651a7e7c901.js:362467:31)
    at beginWork$$1 (http://localhost:9000/static/vendors~main-9d6cea73c651a7e7c901.js:385230:7)
    at performUnitOfWork (http://localhost:9000/static/vendors~main-9d6cea73c651a7e7c901.js:384221:12)
    at workLoopSync (http://localhost:9000/static/vendors~main-9d6cea73c651a7e7c901.js:384198:22): expected false to be true
```

The only way to generate output in headless in `afterEach` is either throwing an Error or failing an assertion.   `cy.task('log',' ...)` which uses `console.log` doesn't output anything :-(